File: DSA/RATNA_KHANNA_590013141/DAY_57_QUESTION_2.c
#include <stdio.h>

int isMaxHeap(int arr[], int n) {
    for (int i = 0; i <= (n - 2) / 2; i++) {
        int l = 2 * i + 1;
        int r = 2 * i + 2;
        if (l < n && arr[i] < arr[l]) return 0;
        if (r < n && arr[i] < arr[r]) return 0;
    }
    return 1;
}

int main() {
    int n;
    printf("enter number of elements\n");
    scanf("%d", &n);
    int arr[n];
    printf("enter elements\n");
    for (int i = 0; i < n; i++) scanf("%d", &arr[i]);
    if (isMaxHeap(arr, n)) printf("true");
    else printf("false");
    return 0;
}

File: DSA/VaniAgrawal_590015846/Day_57_Question_2.py

arr = list(map(int, input("Enter array elements separated by space: ").split()))

def is_max_heap(arr):
    n = len(arr)
    for i in range((n - 2) // 2 + 1):  
        left = 2 * i + 1
        right = 2 * i + 2
        
        if left < n and arr[i] < arr[left]:
            return False
        if right < n and arr[i] < arr[right]:
            return False
    return True

print("true" if is_max_heap(arr) else "false")
